packages/components/tooltip/src/tooltip.tsx

```diff
@@ -156,53 +170,44 @@ const Tooltip = (props: TTooltipProps) => {
 
   useEffect(() => {
     return () => {
+      if (enterTimer.current) {
+        clearTimeout(enterTimer.current);
+      }
       if (leaveTimer.current) {
         clearTimeout(leaveTimer.current);
       }
     };
   }, []);
 
-  const { reference, popper } = usePopper({
+  const { reference, popper, popperInstance } = usePopper({
     placement: props.placement,
     modifiers: props.modifiers,
   });
-  const [isOpen, toggle] = useToggleState(false);
-  const closeTooltip = useCallback(() => {
-    toggle(false);
-  }, [toggle]);
-  const openTooltip = useCallback(() => {
-    toggle(true);
-  }, [toggle]);
+  const [state, setState] = useState<TTooltipState>('closed');
 
   const isControlled = !isNil(props.isOpen);
-  const tooltipIsOpen = isControlled ? props.isOpen : isOpen;
+  const tooltipIsOpen = isControlled
+    ? props.isOpen
+    : state === 'opened' || state === 'exiting';
   const id = useFieldId(props.id, sequentialId);
 
   const { onClose } = props;
   const handleClose = useCallback(
     (event?: ChangeEvent | FocusEvent) => {
       if (!isControlled) {
-        closeTooltip();
+        setState('closed');
       }
       if (onClose) {
         onClose(event);
       }
     },
-    [isControlled, closeTooltip, onClose]
+    [isControlled, onClose]
   );
 
   const { onFocus, onMouseOver } = props.children.props;
-  const { onOpen } = props;
+  const { showAfter, onOpen } = props;
   const handleEnter = useCallback(
     (event?: ChangeEvent | FocusEvent) => {
-      // Remove the title ahead of time.
-      // We don't want to wait for the next render commit.
-      // We would risk displaying two tooltips at the same time (native + this one).
-      if (childrenRef && typeof childrenRef === 'function') {
```

Comment:
`childrenRef` was not assigned to any element so this wasn't working.
Handling the children title is done down below.